pkg/dashboard/server/flux.go

```diff
@@ -101,3 +114,44 @@ func parseRev(rev string) (string, error) {
 
 	return parts[1], nil
 }
+
+func updateGitopsCommitStatuses(
+	gitopsRepoCache *nativeGit.RepoCache,
+	perf *prometheus.HistogramVec,
+	store *store.Store,
+	repoName, env, eventMessage string,
+	repoPerEnv bool,
+) error {
+	since := time.Now().Add(-24 * time.Hour)
+	repo, pathToCleanUp, err := gitopsRepoCache.InstanceForWrite(repoName) // using a copy of the repo to avoid concurrent map writes error
+	defer gitopsRepoCache.CleanupWrittenRepo(pathToCleanUp)
+	if err != nil {
+		return err
+	}
+
+	releases, err := gitops.Releases(repo, "", env, repoPerEnv, &since, nil, 10, "", perf)
+	if err != nil {
+		return err
+	}
+
+	for _, r := range releases {
+		gitopsCommit, err := store.GitopsCommit(r.GitopsRef)
+		if err != nil {
+			log.Warnf("cannot get gitops commit: %s", err)
+			continue
+		}
+		if gitopsCommit == nil || gitopsCommit.Status == dx.NotReconciled {
+			_, err := store.SaveOrUpdateGitopsCommit(&model.GitopsCommit{
+				Sha:        r.GitopsRef,
+				Status:     dx.ReconciliationSucceeded,
```

Comment:
Are you sure Flux sent a succeeded message?
It could happen that Flux applies the last commit out of three and it runs onto an error. Then the state is error that we should record to all the `dx.NotReconciled` commits.

Also this code is not stopping to update commits in case of the following sequence in our db:
dx.NotReconciled 
dx.NotReconciled 
dx.NotReconciled 
dx ReconciliationSucceeded
dx.ReconciliationSucceeded
dx.Failed
dx.NotReconciled

It should really only look back until the last definite state (success or error). Not 10 commits, or 24hrs.